base/compiler/types.jl

```diff
@@ -292,7 +292,7 @@ infer_compilation_signature(::NativeInterpreter) = true
 
 typeinf_lattice(::AbstractInterpreter) = InferenceLattice(BaseInferenceLattice.instance)
 ipo_lattice(::AbstractInterpreter) = InferenceLattice(IPOResultLattice.instance)
-optimizer_lattice(::AbstractInterpreter) = OptimizerLattice()
+optimizer_lattice(::AbstractInterpreter) = OptimizerLattice(IPOResultLattice.instance)
```

Comment:
Currently most parts of our optimization passes don't have things that need to handle `InterConditional` as it only appears within `CallInfo`s, so this may incur (mostly) unnecessary computational costs.

Reply:
Yeah, this should probably be `BaseInferenceLattice`